community/licensecheck-config/src/test/java/org/neo4j/licensecheck/LicenseYearTest.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.licensecheck;
+
+import org.junit.Test;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.net.URL;
+import java.util.Calendar;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+public class LicenseYearTest
+{
+
+    private static final String NOTICE_GPL_FILE = "notice-gpl-prefix.txt";
+    private static final String NOTICE_AGPL_FILE = "notice-agpl-prefix.txt";
+
+    @Test
+    public void testNoticePrefixYearGPL() throws FileNotFoundException
+    {
+        URL resource = getClass().getClassLoader().getResource( NOTICE_GPL_FILE );
+        File gplFile = new File( resource.getFile() );
+
+        checkYearInFile(gplFile);
+    }
+
+    @Test
+    public void testNoticePrefixYearAGPL() throws FileNotFoundException
+    {
+        URL resource = getClass().getClassLoader().getResource( NOTICE_AGPL_FILE );
+        File gplFile = new File( resource.getFile() );
+
+        checkYearInFile( gplFile );
+    }
```

Comment:
Is there some "official build" parameter flag we could use? Which would be yes for packaging builds, but generally false anywhere else.